### PR TITLE
Add missing guide to documentation

### PR DESCRIPTION
**Pull request by ashishb-solo**
# Description

The documentation is currently missing [this page](https://docs.solo.io/gloo-edge/1.11.37/guides/dev/building-and-deploying-gloo/). This pull request is an attempt to implement that missing documentation page

# Context

Some new developers to Gloo found this page to be missing

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/7301

### CODE REVIEW

File: docs/content/guides/dev/building-and-deploying-gloo.md [hunk 1]
```diff
@@ -3,6 +3,115 @@ title: "Building and Deploying Gloo Edge from Source"
...
+
+    make -B gloo
+
+This will output a `gloo` binary in `_output/projects/gloo/`. To build the docker image, run
+
+    make -B gloo-docker
+
+This will build a docker image and tag it as something like this:
+
+    Successfully tagged quay.io/solo-io/gloo:1.13.0-beta13-2-g24ba2e1b8-dirty
```
[the remark this answers is not in the document]
Comment (ashishb-solo) [reply docs/content/guides/dev/building-and-deploying-gloo.md]: addressed here: https://github.com/solo-io/gloo/pull/7295/commits/c82165765ed7018c6c17b833d82c88ac07aac06f

File: docs/content/guides/dev/building-and-deploying-gloo.md [hunk 2]
```diff
@@ -3,6 +3,115 @@ title: "Building and Deploying Gloo Edge from Source"
...
+
+Gloo can be set up and installed in [many different ways](https://docs.solo.io/gloo-edge/latest/installation/preparation/#deployment-requirements). From this point, we will assume we want to deploy Gloo into a kubernetes cluster running locally via [kind](https://docs.solo.io/gloo-edge/latest/installation/platform_configuration/cluster_setup/#kind).
+
+First, install `gloo` into the cluster if it is not already present:
+
+    glooctl install gateway
+
+Then copy the image into the kind cluster:
+
+    kind load docker-image quay.io/solo-io/gloo:1.13.0-beta13-2-g24ba2e1b8-dirty
```
[the remark this answers is not in the document]
Comment (ashishb-solo) [reply docs/content/guides/dev/building-and-deploying-gloo.md]: addressed here: https://github.com/solo-io/gloo/pull/7295/commits/c82165765ed7018c6c17b833d82c88ac07aac06f

File: docs/content/guides/dev/building-and-deploying-gloo.md [hunk 3]
```diff
@@ -3,6 +3,117 @@ title: "Building and Deploying Gloo Edge from Source"
...
+
+After a few seconds, the old pod is deleted:
+
+    $ kubectl -n gloo-system get pods
+    NAME                            READY   STATUS    RESTARTS   AGE
+    discovery-77d7cd499d-vg2rl      1/1     Running   0          89s
+    gateway-proxy-5b88d8995-hzt27   1/1     Running   0          89s
+    gloo-bdd7ddb9d-l977c            1/1     Running   0          16s
+
+Note that it's also easier to observe this update happening in real time by using a live monitoring tool like [k9s](https://k9scli.io/).
```
Comment (jackstine) [inline docs/content/guides/dev/building-and-deploying-gloo.md:73]: Nice 🚀

**Review by jackstine — COMMENTED** [review]
Looking good, I think adding support for m1 would work nicely too.

File: docs/content/guides/dev/building-and-deploying-gloo.md [hunk 4]
```diff
@@ -3,6 +3,131 @@ title: "Building and Deploying Gloo Edge from Source"
...
+
+Next, make the image accessible to the kind cluster. On `x86_64`:
+
+    kind load docker-image quay.io/solo-io/gloo:0.0.1
+
+Or on `arm64` or `m1`:
+
+    docker push localhost:5000/gloo:1.13.0-beta13-2-g24ba2e1b8-dirty
+
+Then update the kind cluster to use this new image as the template. On `x64_64`, replace `image-tag` with `quay.io/solo-io/gloo:0.0.1`, and on `arm64`, replace it with `localhost:5000/solo-io/gloo:0.0.1`
```
Comment (jackstine) [inline docs/content/guides/dev/building-and-deploying-gloo.md:66]:
```suggestion
    kind load docker-image quay.io/solo-io/gloo:0.0.1

Or on `arm64` or `m1`:

    docker push localhost:5000/gloo:0.0.1

Then update the kind cluster to use this new image as the template. On `x64_64`, replace `image-tag` with `quay.io/solo-io/gloo:0.0.1`, and on `arm64`, replace it with `localhost:5000/solo-io/gloo:0.0.1`
```

**Review by jackstine — APPROVED** [review]
Nice!

**Review by Bslabe123 — APPROVED** [review]

**Pull request merged**